data/sea/42-text-conversion.h

```diff
@@ -0,0 +1,357 @@
+#include "41-grisu2.h"
+
+/*
+utils
+*/
+
+static bool INLINE text_is_digit (char c)
+{
+    return c >= '0' && c <= '9';
+}
+
+
+/*
+iint_t read/write
+*/
+
+static const size_t text_iint_max_size = 20; /* 64-bit integer = 19 digits + sign */
+
+static ierror_msg_t INLINE text_read_iint (char **pp, char *pe, iint_t *output_ptr)
+{
+    char *p            = *pp;
+    size_t buffer_size = pe - p;
+
+    size_t n = 0;
+    while (n < buffer_size) {
+        if (!text_is_digit (p[n]))
+            break;
+        n++;
+    }
+
+    if (n == 0)
+        return ierror_msg_alloc ("not an integer", p, buffer_size);
+
+    iint_t value = 0;
+
+    /* handle negative */
+    int sign = 1;
+    if (p[0] == '-') {
+        sign = -1;
+        ++p;
+        --n;
+    }
+
+    /* handle up to 19 digits, assume we're 64-bit */
+    switch (n) {
+        case 19:  value += (p[n-19] - '0') * 1000000000000000000;
+        case 18:  value += (p[n-18] - '0') * 100000000000000000;
+        case 17:  value += (p[n-17] - '0') * 10000000000000000;
+        case 16:  value += (p[n-16] - '0') * 1000000000000000;
+        case 15:  value += (p[n-15] - '0') * 100000000000000;
+        case 14:  value += (p[n-14] - '0') * 10000000000000;
+        case 13:  value += (p[n-13] - '0') * 1000000000000;
+        case 12:  value += (p[n-12] - '0') * 100000000000;
+        case 11:  value += (p[n-11] - '0') * 10000000000;
+        case 10:  value += (p[n-10] - '0') * 1000000000;
+        case  9:  value += (p[n- 9] - '0') * 100000000;
+        case  8:  value += (p[n- 8] - '0') * 10000000;
+        case  7:  value += (p[n- 7] - '0') * 1000000;
+        case  6:  value += (p[n- 6] - '0') * 100000;
+        case  5:  value += (p[n- 5] - '0') * 10000;
+        case  4:  value += (p[n- 4] - '0') * 1000;
+        case  3:  value += (p[n- 3] - '0') * 100;
+        case  2:  value += (p[n- 2] - '0') * 10;
+        case  1:  value += (p[n- 1] - '0');
+                  value *= sign;
+                  *output_ptr = value;
+                  *pp += n;
+                  return 0;
+
+        default:  return ierror_msg_alloc ("integer too big", p, n);
+    }
+}
```

Comment:
This new `text_read_iint` speeds up the ASX 500MB dataset from 410MB/s -> 630MB/s on my computer.